src/osvr/Client/GestureRemoteFactory.cpp

```diff
@@ -0,0 +1,161 @@
+#include <osvr/Util/UniquePtr.h>
+#include <osvr/Common/OriginalSource.h>
+#include "InterfaceTree.h"
+#include <osvr/Util/Verbosity.h>
+#include <osvr/Common/CreateDevice.h>
+#include <osvr/Common/GestureComponent.h>
+#include <osvr/Common/RegisteredStringMap.h>
+
+// Library/third-party includes
+// - none
+
+// Standard includes
+// - none
+
+namespace osvr {
+namespace client {
+
+    class NetworkGestureRemoteHandler : public RemoteHandler {
+      public:
+        NetworkGestureRemoteHandler(vrpn_ConnectionPtr const &conn,
+                                    std::string const &deviceName,
+                                    boost::optional<OSVR_ChannelCount> sensor,
+                                    common::InterfaceList &ifaces)
+            : m_dev(common::createClientDevice(deviceName, conn)),
+              m_interfaces(ifaces), m_all(!sensor.is_initialized()),
+              m_sensor(sensor) {
+
+            auto gesture = common::GestureComponent::create();
+            m_dev->addComponent(gesture);
+
+            gesture->registerGestureHandler(
+                [&](common::GestureData const &data,
+                    util::time::TimeValue const &timestamp) {
+                    m_handleGesture(data, timestamp);
+                });
+            gesture->registerGestureMapHandler(
+                [&](common::GestureMap const &dataMap,
+                    util::time::TimeValue const &timestamp) {
+                    m_handleGestureMap(dataMap, timestamp);
+                });
+
+            /**/
+            OSVR_DEV_VERBOSE("Constructed an Gesture Handler for "
+                             << deviceName);
+        }
+
+        /// @brief Deleted assignment operator.
+        NetworkGestureRemoteHandler &
+        operator=(NetworkGestureRemoteHandler const &) = delete;
+
+        virtual ~NetworkGestureRemoteHandler() {
+            /// @todo do we need to unregister?
+        }
+
+        virtual void update() { m_dev->update(); }
+
+      private:
+        void m_handleGesture(common::GestureData const &data,
+                             util::time::TimeValue const &timestamp) {
+            if (!m_all && *m_sensor != data.sensor) {
+                /// doesn't match our filter.
+                return;
+            }
+
+            OSVR_GestureReport report;
+
+            osvr::common::StringID id =
+                localGestureMap.convertPeerToLocalID(data.gestureID);
+            if (id.empty()) {
+                // could not find a peer to local mapping, discarding report
+                return;
+            }
+            std::string gestureName = localGestureMap.getNameFromID(id);
+
+            if (gestureName.empty()) {
+                // could not find gesture name for this id, discarding report
+                return;
+            }
+            report.sensor = data.sensor;
+            report.state = data.gestureState;
+            report.gestureName = gestureName.c_str();
```

Comment:
Can't do this - c_str() returns a temporary, so you can't expect it to hang around.